subprojects/shipkit/src/main/groovy/org/shipkit/internal/util/GitHubStatusCheck.java

```diff
@@ -0,0 +1,83 @@
+package org.shipkit.internal.util;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.json.simple.JsonObject;
+import org.json.simple.Jsoner;
+import org.shipkit.internal.gradle.versionupgrade.MergePullRequestTask;
+
+import java.io.IOException;
+import java.util.Collection;
+
+public class GitHubStatusCheck {
+
+    private static final Logger LOG = Logging.getLogger(GitHubStatusCheck.class);
+
+    private MergePullRequestTask task;
+    private GitHubApi gitHubApi;
+
+    public GitHubStatusCheck(MergePullRequestTask task, GitHubApi gitHubApi) {
+        this.task = task;
+        this.gitHubApi = gitHubApi;
+    }
+
+    public boolean checkStatusWithTimeout() throws IOException, InterruptedException {
+        int timeouts = 0;
+        while (timeouts < 20) {
+            if (statusCheck(task, gitHubApi)) {
+                return true;
+            } else {
+                int waitTime = 10000 * timeouts;
```

Comment:
@wwilk I'm a bit relucant with that myself but for different case - this solution is not configurable. I didn't wanted to make it more complex than necessary, but I wonder if there won't be case when all checks will be very long. That's why I made waitTime growning in every iteration - still similar solution can be achieved by allowing more retries OR making it configurable (for now it is hardcoded on 20). Please let me know what you think.

Reply:
It will be certainly configurable in the long run, for now it can be a fixed value. We will test it on Shipkit and example project for a while before making it public. We can add a TODO in code. 

I think we can make a little more retries, since 20 * 10 s may be not enough for all shipkit checks to pass. It seems like it's 5-7 minutes at the time. We can set retries = 100 for now. What's the advantage of incremental delay between retries in this case?